File: ws.py
##
##	[1]. Create workspace for the project.
##	Author: Nguyen Chinh Thuy.
##	Date  : 17/08/2017.
##


##
## Import
##
import cv2
from matplotlib.pyplot import *


##
## Function : Read eye image.
##
## Input    : fname : Name of file eye image.
##			  dbpath: Path of sample database.
##
## Output   : im    : Read image.
##            imsz  : Size of the image.
##
def readim(fname, dbpath):
	im = cv2.imread(dbpath + fname, 0)
	imsz = [im.shape[0], im.shape[1]]
	return im, imsz


##
## Function : Plot read image.
##
## Input    : im    : Read image.
##            imsz  : Size of the image.
##            fname : Name of file eye image.
##
## Output   : None.
##
def vs(im, imsz, fname):
	figure(1)
	title('[%s].Input image [%dx%d]' % (fname[0:8],imsz[0],imsz[1]))
	imshow(im, cmap='gray', interpolation='bicubic')
	show()

File: iris-matlab/ws.m
%
%	[1]. Create workspace for the project.
%	Author: Nguyen Chinh Thuy.
%	Date  : 27/07/2017.
%
%	Input : Ws 		: Struct Ws from [exe.m]
%
%	Output: im 		: Eye image.
%			imsz 	: Size of the image. 
%

tic
%% Read image
im = imread(Ws.fname);
if ndims(im)==3
	im = rgb2gray(im);
end
imsz = [size(im,1), size(im,2)];

time.ws=toc;
%% Plot
if(Ws.isPlot)
	figure(1); clf;
	title(sprintf('[%s].Input image [%dx%d]',Ws.fname(1:8),imsz(1),imsz(2)));
	imshow(im);
% 	surf(im(1:8:imsz(1), 1:8:imsz(2))); rotate3d on; view(90,90);
end

